contract/r/gnoswap/gnft/gnft_test.gno

```diff
@@ -172,49 +172,71 @@ func TestIsApprovedForAll(t *testing.T) {
 func TestSetTokenURI(t *testing.T) {
 	tests := []struct {
 		name             string
+		setup            func()
 		callerRealm      runtime.Realm
 		tokenId          uint64
 		expectedHasAbort bool
 		expectedErrMsg   string
 	}{
 		{
-			name:             "set token uri expected error",
-			callerRealm:      addr01Realm,
+			name: "successfully set token uri by owner",
+			setup: func() {
+				resetObject(t)
+				testing.SetRealm(positionRealm)
+				// Mint token without URI by directly calling nft.Mint
+				ownerAddress := owner.Owner()
+				checkErr(nft.Mint(ownerAddress, tid(1)))
+			},
+			callerRealm:      positionRealm,
 			tokenId:          1,
```

Comment:
You could consider passing it as a parameter to the `setup` callback since the test data includes a `tokenId` field.

Reply:
I've made the changes! And I added some test cases.